tests/unit/v1/test_async_collection.py

```diff
@@ -206,17 +206,19 @@ def _next_page(self):
         documents = [
             Document(name=template.format(document_id)) for document_id in document_ids
         ]
-        iterator = _Iterator(pages=[documents])
+        iterator = _AsyncIterator(pages=[documents])
         firestore_api = AsyncMock()
         firestore_api.mock_add_spec(spec=["list_documents"])
         firestore_api.list_documents.return_value = iterator
         client._firestore_api_internal = firestore_api
         collection = self._make_one("collection", client=client)
 
         if page_size is not None:
-            documents = list(await collection.list_documents(page_size=page_size))
+            documents = [
+                i async for i in collection.list_documents(page_size=page_size)
+            ]
         else:
-            documents = list(await collection.list_documents())
+            documents = [i async for i in collection.list_documents()]
```

Comment:
Is this change required? It seems at first glance it should be equivalent?

Reply:
You would think so, but `list()` can't be used to create a list from an async generator, as async generators are not iterable. Unfortunately, this is the preferred solution to creating a list from an async generator.

Reply:
If we want to use something like `list(await collection.list_documents())` we could refactor `list_documents` to return a list, rather than an async generator. However, we would lose on the benefit of a generator being able to incrementally return results as they arrive, rather than compiling all results before returning.